holidays/countries/spain.py

```diff
@@ -53,70 +53,77 @@ def _is_observed(self, date_holiday, name_holiday):
     def _populate(self, year):
         super()._populate(year)
 
-        self._is_observed(date(year, JAN, 1), "Año nuevo")
+        if year == 2023 and self.subdiv in {
+            "AN",
+            "AR",
+            "AS",
+            "CL",
+            "MC",
+        }:
+            self._is_observed(date(year, JAN, 1), "Año nuevo")
```

Comment:
In rest of regions there was no New Year holiday at all?

Reply:
Since 1th of January was Sunday, just some regions observed this festivity the day after.

Reply:
For reference:

https://administracion.gob.es/pag_Home/dam/jcr:a31ce637-a7ba-49e1-9f41-e7a86d8ec867/Calendario_Laboral2023.pdf

Reply:
it's my understanding that when a holiday happens on Sunday it's not added
to the list of holidays for that year.


For reference:

https://administracion.gob.es/pag_Home/dam/jcr:a31ce637-a7ba-49e1-9f41-e7a86d8ec867/Calendario_Laboral2023.pdf

Reply:
I believe it's worth adding `https://administracion.gob.es/pag_Home/atencionCiudadana/calendarios.html` to Spain docstring as a reference.